pkg/controller/migmigration/directimagemigration.go

```diff
@@ -88,3 +109,37 @@ func (t *Task) setDirectImageMigrationWarning(dim *migapi.DirectImageMigration)
 		})
 	}
 }
+
+func (t *Task) deleteDirectImageMigrationResources() error {
+	// fetch the destination cluster client
+	destClient, err := t.getDestinationClient()
```

Comment:
We should not assume that the DIM resource lives on the target cluster, as this will not always be the case (e.g. if controller is running on src cluster). We should instead use the controller clusters client, `t.Client`

Reply:
Done, updated the code.